Extensions/AdMob/admobtools.ts

```diff
@@ -93,288 +172,415 @@ namespace gdjs {
-      return interstitialLoading;
-    };
-    /** Check if the interstitial is ready to display. */
-    export const isInterstitialReady = () => {
-      return interstitialReady;
-    };
-    /** Check if the interstitial is shown on screen. */
-    export const isInterstitialShowing = () => {
-      return interstitialShowing;
-    };
-    /** Check if the interstitial had an error while loading it. */
-    export const isInterstitialErrored = () => {
-      return interstitialErrored;
-    };
+    export const isInterstitialLoading = () => interstitialLoading;
+    export const isInterstitialReady = () => interstitialReady;
+    export const isInterstitialShowing = () => interstitialShowing;
+    export const isInterstitialErrored = () => interstitialErrored;
 
     /** Load an interstitial. */
-    export const loadInterstitial = (androidID, iosID, displayWhenLoaded) => {
-      if (typeof admob === 'undefined') {
-        return;
-      }
-      if (interstitialLoading || interstitialReady || interstitialShowing) {
+    export const loadInterstitial = async (
+      androidAdUnitId,
+      iosAdUnitId,
+      displayWhenLoaded
+    ) => {
+      if (!checkIfAdMobIsAvailable()) return;
+      // If an interstitial is already loading or showing, we don't stop it.
+      if (interstitialLoading || interstitialShowing) {
         return;
       }
 
+      const adUnitId = getAdUnitId(
+        androidAdUnitId,
+        iosAdUnitId,
+        'interstitial'
+      );
+      if (!adUnitId) return;
+
       interstitialLoading = true;
       interstitialReady = false;
       interstitialErrored = false;
-      admob.interstitial
-        .load({
-          id: {
-            android: androidID,
-            ios: iosID,
-          },
-          npa: npaValue,
-        })
-        .then(
-          () => {
-            logger.info('AdMob interstitial successfully loaded.');
-            if (displayWhenLoaded) showInterstitial();
-          },
-          (error) => {
-            interstitialLoading = false;
-            interstitialReady = false;
-            interstitialErrored = true;
-            logger.error('Error while loading a interstitial:', error);
-          }
-        );
+
+      interstitial = new admob.InterstitialAd({
+        adUnitId,
+        npa: npaValue,
+      });
+
+      interstitial.on('load', () => {
+        interstitialReady = true;
+        interstitialLoading = false;
+      });
+      interstitial.on('loadfail', () => {
+        interstitialLoading = false;
+        interstitialErrored = true;
+      });
+      interstitial.on('show', () => {
+        interstitialShowing = true;
+        interstitialReady = false;
+      });
+      interstitial.on('showfail', () => {
+        interstitialShowing = false;
+        interstitialErrored = true;
+      });
+      interstitial.on('dismiss', () => {
+        interstitialShowing = false;
+      });
+
+      try {
+        logger.info('loading Admob interstitial.....');
```

Comment:
Is this for debugging? If not it should be Capitalized + single period.

Reply:
I'll update all of them, I think it's valuable to keep the logs